cmd/pomerium/main.go

```diff
@@ -102,56 +103,44 @@ func startRedirectServer(addr string) (*http.Server, error) {
 	return srv, nil
 }
 
-func newAuthenticateService(s string, mux *http.ServeMux, rpc *grpc.Server) (*authenticate.Authenticate, error) {
-	if !isAuthenticate(s) {
+func newAuthenticateService(opt *config.Options, mux *http.ServeMux, rpc *grpc.Server) (*authenticate.Authenticate, error) {
+	if !config.IsAuthenticate(opt.Services) {
 		return nil, nil
 	}
-	opts, err := authenticate.OptionsFromEnvConfig()
-	if err != nil {
-		return nil, err
-	}
-	service, err := authenticate.New(opts)
+	service, err := authenticate.New(opt)
 	if err != nil {
 		return nil, err
 	}
 	pbAuthenticate.RegisterAuthenticatorServer(rpc, service)
-	mux.Handle(urlutil.StripPort(opts.AuthenticateURL.Host)+"/", service.Handler())
+	mux.Handle(urlutil.StripPort(opt.AuthenticateURL.Host)+"/", service.Handler())
 	return service, nil
 }
 
-func newAuthorizeService(s string, rpc *grpc.Server) (*authorize.Authorize, error) {
-	if !isAuthorize(s) {
+func newAuthorizeService(opt *config.Options, rpc *grpc.Server) (*authorize.Authorize, error) {
+	if !config.IsAuthorize(opt.Services) {
```

Comment:
Could panic if opt is nil?

Reply:
Yes.  Argument to be made that we've really screw up in main() if that happens.  I'll add a check in, though.